parity/configuration.rs

```diff
@@ -1234,11 +1239,11 @@ struct UiEnabled {
 	pub info_page_only: bool,
 }
 
-fn into_secretstore_service_contract_address(s: &str) -> Result<Option<SecretStoreContractAddress>, String> {
-	match s {
-		"none" => Ok(None),
-		"registry" => Ok(Some(SecretStoreContractAddress::Registry)),
-		a => Ok(Some(SecretStoreContractAddress::Address(a.parse().map_err(|e| format!("{}", e))?))),
+fn into_secretstore_service_contract_address(s: Option<&String>) -> Result<Option<SecretStoreContractAddress>, String> {
+	match s.map(|x| &**x) {
```

Comment:
I think this should be simplified to `s.map(String::as_str)` for readability purposes!